docs/v1/P4Runtime-Spec.mdk

```diff
@@ -6269,6 +6279,356 @@ field [@ProtoAny]. At the moment, there is not any mechanism to extend the
 `p4.v1.TableEntry` message based on the value of architecture-specific table
 properties, but we may include on in future versions of the API.
 
+# GenericTable { #sec-generic-table}
+
+## GenericTable p4info { #sec-p4info-generic-table}
+
+`GenericTable` messages are used to map non-PSA P4-externs or non-P4 target-
+specific "fixed" features to their implementation in a generic way. The same can be achieved
+via `Extern`, but GenericTable provides a structured way in which every feature is
+represented as a set of match-fields and data-fields. The data consists of unions
+which are similar to actions. One use of GenericTable in a server backend is
+with TDI [Table Driven Interface](https://github.com/p4lang/tdi), but targets
+can use Generic table to map to their own specific feature implementations as
+well.
+It defines the following fields
+
+* `generic_table_type_id`, a 8-bit unsigned integer which uniquely identifies the
+  generic_table_type in the context of the architecture.
+  This value is in the range of `[0x00, 0xff]`. 
+  The ID in the preamble is created as `0x18` `generic_table_type_id` `16 bits of
+  generated ID`.
```

Comment:
I don't understand the sentence above.  What are the 3 quoted things `0x18` `generic_table_type_id` `16 bits of generated ID` ?

Reply:
It took me a while to understand what I had meant by that too. Replacing it with
`0x18 + generic_table_type_id + 16 bits of generated ID by compiler`

Reply:
If by `+` you mean concatenation of bit strings, I would recommend using P4_16 syntax `++` for that, since all readers should be familiar with it.

Reply:
Never mind my previous comment.  I see you have used precise C/C++/P4_16 syntax that all readers should be familiar with.  Thanks.